src/conn/conn_prefetch.c

```diff
@@ -182,6 +182,16 @@ __wt_conn_prefetch_clear_tree(WT_SESSION_IMPL *session, bool all)
       "Pre-fetch needs to save a valid dhandle when clearing the queue for a btree");
 
     __wt_spin_lock(session, &conn->prefetch_lock);
+    /*
+     * In order for this to be set the thread fetching the pages must have taken and released the
```

Comment:
We guarantee that no dhandle enters the prefetch busy state while we wait. This is because we hold the lock while draining, and the lock is required when taking from the queue.